docs/migrating-from-0.5.md

```diff
@@ -2,7 +2,118 @@
 
 Genkit 0.9 introduces a number of breaking changes alongside feature enhancements that improve overall functionality. If you have been developing applications with Genkit 0.5, you will need to update your application code when you upgrade to the latest version. This guide outlines the most significant changes and offers steps to migrate your existing applications smoothly.
 
-## 1. CLI Changes
+## Quickstart guide
+
+The following steps will get you up and running on Genkit 0.9 quickly. Read more information about these changes in the detailed [Changelog](#changelog) below.
+
+### 1. Install the new CLI
+
+* Uninstall the old CLI
+
+```posix-terminal
+npm uninstall -g genkit
+```
+
+* Install the new CLI
+
+```posix-terminal
+npm i -g genkit-cli
+```
+
+### 2. Update your dependencies
+
+* Remove separated Genkit core packages
+
+```posix-terminal
+npm uninstall @genkit-ai/ai @genkit-ai/core @genkit-ai/dotprompt @genkit-ai/flow
+```
+
+* Install the new consolidated `genkit` package
+
+```posix-terminal
+npm i --save genkit
+```
+
+* Upgrade all plugin versions (example below)
+
+```
+npm upgrade @genkit-ai/firebase
+```
+
+### 3. Change your imports
+
+* Remove imports from separated Genkit core packages
+
+```js
+import { … } from '@genkit-ai/ai';
+import { … } from '@genkit-ai/core';
+import { … } from '@genkit-ai/flow';
+import * as z from 'zod';
+```
+  
+* Import `genkit` and `zod` from `genkit`
+
+```js
+import { z, genkit, defineFlow,  } from 'genkit';
+```
+
+### 4. Update your code
+
+#### Configuration is now much simpler 
+
+Configuration for genkit is now done per instance. Telemetry and logging is configured globally and separately from the genkit instance. 
+
+See the [Monitoring and Logging](./monitoring.md) documentation for more details on how to configure telemetry and logging.
+
+See the [Get Started](./get-started.md) documentation for more details on how to configure a Genkit instance.
+
+```js
+import { genkit } from 'genkit';
+import { logger } from 'genkit/logging';
+import { enableFirebaseTelemetry } from '@genkit-ai/firebase';
+
+logger.setLogLevel('debug');
```

Comment:
is this necessary?

Reply:
I wonder if `enableFirebaseTelemetry` could do it?

Reply:
feel free to ignore here... something to consider.

Reply:
It's a good question. Right now the logger/logging transports for Firebase telemetry is separate from the logger used by core genkit. In the case where you aren't deployed on Google Cloud infrastructure, I think this might make a big difference.